src/AutoMapper/TypeMap.cs

```diff
@@ -328,4 +338,39 @@ private void ApplyInheritedTypeMap(TypeMap inheritedTypeMap)
             }
         }
     }
+
+    public class TypeMap<TSource, TDestination>

```

Comment:
Yes cause inlining the expression will show everything it calls.  But in a lot of the code you translate expressions to invoke functions the user passes to configure it. [Before Map](https://github.com/AutoMapper/AutoMapper/blob/2cb87ad30977f34ddccac9e7bac407893f1626b7/src/AutoMapper/Configuration/MappingExpression.cs#L348) shows this issue cause your expression generated is calling an action that turns into a constant value when you translate it.  And that constant won't be able to be translated into `CompileToMethod().

If you remove CanCompile visitor check when making the delegate and run the tests you can see all the exceptions for yourself.